src/improvements/AddressRegistry.sol

```diff
@@ -410,34 +410,30 @@ contract AddressRegistry is IAddressRegistry, Test {
 
     function getAnchorStateRegistryProxy(address faultDisputeGame, address permissionedDisputeGame)
         internal
-        view
         returns (address)
     {
-        try IFetcher(faultDisputeGame).anchorStateRegistry() returns (address anchorStateRegistryProxy) {
-            return anchorStateRegistryProxy;
-        } catch {
-            return IFetcher(permissionedDisputeGame).anchorStateRegistry();
-        }
+        (bool ok, bytes memory data) =
+            address(faultDisputeGame).call(abi.encodeWithSelector(IFetcher.anchorStateRegistry.selector));
+        if (ok && data.length == 32) return abi.decode(data, (address));
+
+        return IFetcher(permissionedDisputeGame).anchorStateRegistry();
```

Comment:
There's an invariant that ALL chains on any fault proof version will have a permissioned dispute game. And the fault dispute game is only present for chains on permissionless proofs
```suggestion
        return IFetcher(permissionedDisputeGame).anchorStateRegistry();
```